app/GNSSStatus/Parsing/GNSSData.cs

```diff
@@ -35,13 +49,49 @@ public string GetPayloadJson()
         double roverXAverage = RoverXCache.Count > 0 ? RoverXCache.Average() : 0;
         double roverYAverage = RoverYCache.Count > 0 ? RoverYCache.Average() : 0;
         double roverZAverage = RoverZCache.Count > 0 ? RoverZCache.Average() : 0;
+        */
+        
+        // Calculate median and clear caches.
+        int medianIndex = CalculateMedianIndex(RoverZCache).Item1;
+        int medianPdop = CalculateMedianIndex(RoverPDopCache).Item1;
+        int medianGstAltitudeError = CalculateMedianIndex(RoverErrorAltitudeCache).Item1;
+        int medianBaseline = CalculateMedianIndex(RoverBaselineCache).Item1;
+        double roverXMedian = RoverXCache[medianIndex];
+        double roverYMedian = RoverYCache[medianIndex];
+        double roverZMedian = RoverZCache[medianIndex];
+        double deltaXMedian = DeltaXCache[medianIndex];
+        double deltaYMedian = DeltaYCache[medianIndex];
+        double deltaXy = Math.Sqrt(deltaXMedian * deltaXMedian + deltaYMedian * deltaYMedian);
+        string roverUtCTime = RoverUtcTimeCache[medianIndex];
+        GGAData.FixType roverFixType = RoverFixTypeCache[medianIndex];
+        int roverSatInUse = RoverSatInUseCache[medianIndex];
+        double deltaZ = DeltaZCache[medianIndex];
+        float pDop = RoverPDopCache[medianPdop];
+        float hDop = RoverHDopCache[medianPdop];
+        float vDop = RoverVDopCache[medianPdop];
+        float errLat = RoverErrorLatitudeCache[medianGstAltitudeError];
+        float errLon = RoverErrorLongitudeCache[medianGstAltitudeError];
+        float errAlt = RoverErrorAltitudeCache[medianGstAltitudeError];
+        double baseline = RoverBaselineCache[medianBaseline];
```

Comment:
The same median index cannot be used to index into different data lists, as some lists may contain different amounts of data.